app/gui/view/component-browser/searcher-list-panel/src/layout.rs

```diff
@@ -0,0 +1,340 @@
+//! A module with description how the list of Component Groups is laid out in the Component Panel
+//! List.
+
+use crate::prelude::*;
+
+use ensogl_grid_view::Col;
+use ensogl_grid_view::Row;
+use ide_view_component_group::set::GroupId;
+use ide_view_component_group::set::SectionId;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+/// Height of the header of the component group. This value is added to the group's number of
+/// entries to get the total height.
+pub const HEADER_HEIGHT: usize = 1;
+
+
+
+// =================
+// === ElementId ===
+// =================
+
+/// An identifier of element inside a concrete group: entry (by entry index) or header.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum GroupElement {
+    /// A group's header.
+    Header,
+    /// A group's normal entry with index.
+    Entry(usize),
+}
+
+/// An identifier of some group's element in Component Browser.
+#[allow(missing_docs)]
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub struct ElementId {
+    pub group:   GroupId,
+    pub element: GroupElement,
+}
+
+
+
+// =============
+// === Group ===
+// =============
+
+/// A information about group needed to compute the Component Panel List layout.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct Group {
+    /// The group identifier.
+    pub id:              GroupId,
+    /// Height of group in rows, not counting the header nor filtered-out entries.
+    pub height:          usize,
+    /// Height of group in rows if no entry is filtered out, not counting the header.
+    pub original_height: usize,
+}
+
+
+// === LaidGroup ===
+
+/// The information of group in the layout.
+#[derive(Copy, Clone, Debug)]
+pub struct LaidGroup<'a> {
+    /// Number of row where header is placed.
+    pub header_row: Row,
+    /// The column where the group is placed.
```

Comment:
minor inconsistency in the docs. Make them similar please (either both should tell "the column/row" or both should tell "number of ..."). Btw, "number of" is probably wrong, it should be "index of". "number of" refers to counting things, not indexing them.